core/vm/contract.go

```diff
@@ -177,18 +179,53 @@ func (c *Contract) Value() *big.Int {
 	return c.value
 }
 
+// IsLegacy returns true if contract is not EOF
+func (c *Contract) IsLegacy() bool {
+	// EOF1 doesn't allow contracts without code section
+	return c.header.codeSize == 0
+}
+
+// CodeBeginOffset returns starting offset of the code section
+func (c *Contract) CodeBeginOffset() uint64 {
+	if c.IsLegacy() {
+		return 0
+	}
+	return c.header.CodeBeginOffset()
+}
+
+// CodeEndOffset returns offset of the code section end
+func (c *Contract) CodeEndOffset() uint64 {
+	if c.IsLegacy() {
+		return uint64(len(c.Code))
+	}
+	return c.header.CodeEndOffset()
+}
+
+func (c *Contract) CodeSize() uint64 {
```

Comment:
Docstring. This is actually not just a nit-pick, one pretty important thing to mention here is whether CodeSize returns the size of `header+code`, the entire container, or something else

Reply:
Fixed.